include/superior_mysqlpp/extras/prepared_statement_utils.hpp

```diff
@@ -140,15 +140,19 @@ namespace SuperiorMySqlpp
      * @param ps Prepared statement object (only static statements are currently supported)
      * @param values References to variables to be loaded from query
      *               Their type must be compatible with query result types
-     * @throws std::runtime_error if more than one row is being loaded
+     * @throws UnexpectedZeroRowsError if result is empty
+     * @throws UnexpectedMultipleRowserror if more than one row is being loaded
      */
     template<typename PreparedStatementType, typename... Args, typename = typename std::enable_if<std::is_base_of<detail::StatementBase, std::remove_reference_t<PreparedStatementType>>::value>::type>
     void psReadValues(PreparedStatementType &&ps, Args&... values)
     {
         ps.execute();
 
-        if(ps.getRowsCount() != 1 || !ps.fetch())
+        if (ps.getRowsCount() != 1 || !ps.fetch())
         {
+            if (ps.getRowsCount() == 0) {
+                throw UnexpectedZeroRowsError("psReadValues() has to have a single-row result!");
+            }
```

Comment:
I'd prefer `else` to highlight that the throw is there for negated condition.

Reply:
I don't get it... can you explain by example?

Reply:
Put the other throw in else branch, which by now is already done.

Reply:
But now there is only one exception, isn't it? There is no need for this inner `ps.getRowsCount() == 0)` if

Reply:
Yes, it's not relevant in current version anymore.